util/reggen/countermeasure.py

```diff
@@ -0,0 +1,71 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, Sequence
+
+from .lib import check_keys, check_str, check_list
+
+# TO BE DEFINED
+CM_ASSETS = [
+    'FOO'
+]
+
+# TO BE DEFINED
+CM_TYPES = [
+    'BAR'
+]
+
+
+class CounterMeasure:
+    def __init__(self, name: str, desc: str):
+        self.name = name
+        self.desc = desc
+
+    @staticmethod
+    def from_raw(what: str, raw: object, bname: str) -> 'CounterMeasure':
+        rd = check_keys(raw, what, ['name', 'desc'], [])
+
+        name = check_str(rd['name'], 'name field of ' + what)
+        desc = check_str(rd['desc'], 'desc field of ' + what)
+
+        name_split = name.split('.')
+
+        if len(name_split) != 3:
```

Comment:
Do we need to mandate that the first field be the IP name? IMO it can be inferred from the IP HJson itself.

Reply:
I directly used the naming format defined in the spreadsheet, but you are right that this can be inferred from the IP Hjson (e.g. when building an aggregated top-level list).

I'll remove it.